verifier/src/x509.rs

```diff
@@ -29,6 +35,53 @@ impl From<mbedtls::Error> for Error {
     }
 }
 
+impl From<der::Error> for Error {
+    fn from(src: der::Error) -> Self {
+        Error::Der(src)
+    }
+}
+
+impl From<Error> for CertificateChainVerifierError {
+    fn from(error: Error) -> Self {
+        match error {
+            // Any error (expired, revoked, etc) in certificate chain verification comes back as
+            // `X509CertVerifyFailed`
+            Error::MbedTls(mbedtls::Error::X509CertVerifyFailed) => {
+                CertificateChainVerifierError::SignatureVerification
+            }
+            _ => CertificateChainVerifierError::GeneralCertificateError,
+        }
+    }
+}
+
+/// A certificate chain verifier that uses MbedTls as the backend
+#[derive(Debug)]
+pub struct X509CertificateChainVerifier {
+    trust_anchor: TrustAnchor,
+}
+
+impl X509CertificateChainVerifier {
+    /// Create a new instance
+    pub fn new(trust_anchor: TrustAnchor) -> Self {
+        Self { trust_anchor }
+    }
+}
+
+impl CertificateChainVerifier for X509CertificateChainVerifier {
+    // Note: `_time` is ignored because MbedTls will call out to a system timer.
```

Comment:
Boo, not an issue for this PR to address, but that's a bummer since I think this means we can't run this code inside an enclave. I am not sure that's something we would've needed, I don't have the full picture on what enclave-to-enclave attestation looks like.

Reply:
> since I think this means we can't run this code inside an enclave.

The time in enclaves has to come from untrusted, so it's not trusted anyway :(.
However mbedtls is currently used in the enclaves, it doesn't use time, maybe I need to update the comment:
- https://github.com/mobilecoinfoundation/rust-mbedtls/blob/6d8fe323a3292f87a6bce4b35963d47139a583f9/mbedtls-sys/vendor/library/x509.c#L984 When `MBEDTLS_HAVE_TIME_DATE` is not defined.
- The default value of `MBEDTLS_HAVE_TIME_DATE`, https://github.com/mobilecoinfoundation/rust-mbedtls/blob/6d8fe323a3292f87a6bce4b35963d47139a583f9/mbedtls-sys/build/config.rs#L74
- An option to enable time/date, https://github.com/mobilecoinfoundation/rust-mbedtls/blob/6d8fe323a3292f87a6bce4b35963d47139a583f9/mbedtls-sys/build/config.rs#L420
  - It doesn't appear to be used for `"time"`, https://github.com/mobilecoinfoundation/rust-mbedtls/blob/6d8fe323a3292f87a6bce4b35963d47139a583f9/mbedtls-sys/build/features.rs#L22

Reply:
Updated the comment to say that time is ignored all the way down for mbedtls:(.

Reply:
Makes sense, thanks for the detailed walk-through for the mbedtls implementation. Comment looks good.